packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-customer-section/order-customer-section.tsx

```diff
@@ -0,0 +1,319 @@
+
+  return name
+}
+
+const ID = ({ order }: { order: Order }) => {
+  const { t } = useTranslation()
+
+  const id = order.customer_id
+  const name = getCustomerName(order)
+  const email = order.email
+  const fallback = (name || email).charAt(0).toUpperCase()
+
+  return (
+    <div className="text-ui-fg-subtle grid grid-cols-2 items-center px-6 py-4">
+      <Text size="small" leading="compact" weight="plus">
+        {t("fields.id")}
+      </Text>
+      <Link
+        to={`/customers/${id}`}
+        className="focus:shadow-borders-focus rounded-[4px] outline-none transition-shadow"
+      >
+        <div className="flex items-center gap-x-2 overflow-hidden">
+          <Avatar size="2xsmall" fallback={fallback} />
+          <Text
+            size="small"
+            leading="compact"
+            className="text-ui-fg-subtle hover:text-ui-fg-base transition-fg truncate"
+          >
+            {name || email}
+          </Text>
+        </div>
+      </Link>
+    </div>
+  )
+}
+
+const Company = ({ order }: { order: Order }) => {
+  const { t } = useTranslation()
+  const company =
+    order.shipping_address?.company || order.billing_address?.company
+
+  if (!company) {
+    return null
+  }
+
+  return (
+    <div className="text-ui-fg-subtle grid grid-cols-2 items-center px-6 py-4">
+      <Text size="small" leading="compact" weight="plus">
+        {t("fields.company")}
+      </Text>
+      <Text size="small" leading="compact" className="truncate">
+        {company}
+      </Text>
+    </div>
+  )
+}
+
+const Contact = ({ order }: { order: Order }) => {
+  const { t } = useTranslation()
+
+  const phone = order.shipping_address?.phone || order.billing_address?.phone
+  const email = order.email
+
+  return (
+    <div className="text-ui-fg-subtle grid grid-cols-2 items-start px-6 py-4">
+      <Text size="small" leading="compact" weight="plus">
+        {t("orders.customer.contactLabel")}
+      </Text>
+      <div className="flex flex-col gap-y-2">
+        <div className="grid grid-cols-[1fr_20px] items-start gap-x-2">
+          <Text
+            size="small"
+            leading="compact"
+            className="text-pretty break-all"
+          >
+            {email}
+          </Text>
+
+          <div className="flex justify-end">
+            <Copy content={email} className="text-ui-fg-muted" />
+          </div>
+        </div>
+        {phone && (
+          <div className="grid grid-cols-[1fr_20px] items-start gap-x-2">
+            <Text
+              size="small"
+              leading="compact"
+              className="text-pretty break-all"
+            >
+              {phone}
+            </Text>
+
+            <div className="flex justify-end">
+              <Copy content={email} className="text-ui-fg-muted" />
+            </div>
+          </div>
+        )}
+      </div>
+    </div>
+  )
+}
+
+const isSameAddress = (a: MedusaAddress | null, b: MedusaAddress | null) => {
```

Comment:
suggestion: some of these address utils could be reused - should we move them to `lib/addresses`

Reply:
For sure, but lets hold of with that until we actually need to reuse it somewhere else so we keep global code as lean as possible

Reply:
For sure, but let's hold of with that until we actually need to reuse it somewhere else so we keep global code as lean as possible